src/main/java/lotto/util/LottoNumberGenerator.java

```diff
@@ -0,0 +1,26 @@
+package lotto.util;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import lotto.domain.LottoNumber;
+
+interface LottoNumberGenerator {
+
+	int MINIMUM_NUMBER_BOUND = 1;
+	int MAXIMUM_NUMBER_BOUND = 45;
+	int START_INDEX_OF_NUMBER_ELEMENTS = 0;
+	int NUMBER_OF_NUMBER_ELEMENTS = 6;
```

Comment:
인터페이스에서 상수 사용은 지양해주시면 어떨까요?
다음 링크를 참고하시면 좋을 것 같습니다.

https://stackoverflow.com/questions/2659593/what-is-the-use-of-interface-constants

Reply:
> The constant interface pattern is a poor use of interfaces. That a class uses some constants internally is an implementation detail. Implementing a constant interface causes this implementation detail to leak into the class's exported API. It is of no consequence to the users of a class that the class implements a constant interface. In fact, it may even confuse them. Worse, it represents a commitment: if in a future release the class is modified so that it no longer needs to use the constants, it still must implement the interface to ensure binary compatibility. If a nonfinal class implements a constant interface, all of its subclasses will have their namespaces polluted by the constants in the interface.

이펙티브 자바의 내용이라는데, 위의 주장 논거에 대해 잘 이해가 가지 않습니다. 
1) 구현의 세부사항을 외부의 API로 유출시킬 이유가 없다. 
* 그 이유를 잘 모르겠네요. 
* 반대로 말하면 외부의 API로 유출시킬 필요가 있는가일텐데요. 민감 정보가 아니며, 인터페이스 상수로 구현된다면 관리가 더 쉽지 않을까요. 혹시 구현체 자체에서 상수의 값에 대한 가독성이 떨어져서 그런 것일까요?
2) 구현체가 나중에 바뀌더라도 해당 상수는 남아있을 것이다.
* 음, 인터페이스 상수를 사용한다면 해당 상수를 변경하면 되는 일이 아닐까요?

Reply:
진형님의 의견을 공유해주셔서 감사합니다 👍 
진형님의 의견 또한 많이 공감이 갑니다!  제 의견을 조심스럽게 말씀드려보겠습니다.

반대로 말하면 외부의 API로 유출시킬 필요가 있는가일텐데요. 민감 정보가 아니며, 인터페이스 상수로 구현된다면 관리가 더 쉽지 않을까요. 혹시 구현체 자체에서 상수의 값에 대한 가독성이 떨어져서 그런 것일까요?

-> 인터페이스 상수로 구현된다면 하위 클래스
인터페이스에 대해 고민해보시면 좋을 것 같습니다. 인터페이스를 왜 생성해야될까요?
다형성을 위해서 때문이라고 한다면 절반은 정답이고 절반은 틀렸다고 생각합니다.
인터페이스나 추상클래스나 추상화를 하여 공통점을 추출하고 차이점을 뒤로 감추어
타입을 부여하여 다루는 것이라고 생각해요. 

——————————————
인터페이스에 상수가 존재하게 되면 인터페이스의 구현 객체들은 공통 상수값을 쉽게 가져다 쓸 수 있기 때문에 편할 수 있다고 생각하지만, 잘못된 설계라고 생각해요.

1. 인터페이스의 타입이 정해지는 시점 즉, 다형성을 통해 구체적인 하위 타입이 무엇인지 
정해지는 시점은 런타임입니다.
상수는 런타임이 아닌 클래스 로더가 메모리에 적재하는 시점입니다.
인터페이스의 타입이 정해지는 시점과 상수 코드의 시점은 다르기 때문에 
다형성을 위해서라도 상수를 인터페이스에 사용하는게 맞을지 고민해보시면 좋을 것 같습니다.  

2. 인터페이스는 계약서, 약속입니다. 
해당 타입은 이런 동작을 할것이라고 강제성을 직접적으로 명시하여 약속된 행동을 한다는것을
알려줍니다. 그렇기에 해당 인터페이스를 사용하는 객체들은 구현에 상관없이 계약된, 약속된 행동을 믿고 사용합니다. 상수라는 데이터는 구현에 가깝다고 생각합니다.
그렇기 때문에 인터페이스의 상수를 공개해버리면 외부에서 그 상수를 사용할 수 있게 되는데
만약 해당 인터페이스의 상수가 변경이 된다면 그 상수를 사용하고 있는 외부가 얼마나 사용하고 있는지 추적이 되지 않기 때문에 사이드 이펙트가 엄청날 것 같아요.

예로 현재 저는 포비님과 넥스트스텝을 통해 리뷰를 할 수 있는 관계를 맺고있습니다.
리뷰 기간은 5주로 잡고 계약을 했으므로 모든 리뷰어들에게 공통으로 적용될 수 있게
Reviewable 인터페이스에 REVIEW_PERIOD = 5로 두었다고 가정해보겠습니다.

외부에서 짭스트스텝(?) 이라는 곳이 존재하는데 넥스트스텝을 모방하기 위해 똑같은 구조로
운영하려고 하고 있다고 가정해보겠습니다.
넥스트 스텝의 Reviewable을 통해 ‘아 5주동안 리뷰하는구나’ 정보를 가져와서 사용한다고 했을때

이번 코로나 같이 예측할 수 없는 원인으로 인해 5주에서 8주로 변경이 되었을 경우,
넥스트 스텝에서 자체적으로 처리하던 구현 로직이 있기 때문에 늘어난 만큼 계약금도 자동으로 변경되어 8주로 변경해도 영향이 없지만
외부인 짭스트 스텝에서는 이 리뷰기간이 변경될지 모르고 계약금을 책정했는데
변동됨에 따라 계약금도 변경되어 임금 체불이 될 수도 있지 않을까요?

3. 상수도 민감 정보라고 생각하며 구현에 가까운 정보라고 생각합니다. 
예로 java8 HashMap에서는 기본으로 버킷을 LinkedList로 관리하는데
버킷이 8개 이상이 되면 레드블랙트리로 변경하여 관리하도록 구현되어 있습니다.
다시 버킷이 6개 이하가 된다면  LinkedList로 변경되도록 구현되어있습니다.

6, 8은 상수로 관리하고 있는데, 이 6, 8은 중요하지 않다고 볼 수 있으며,
구현과 관련 없다고 생각이 될까요? 또한 이때까지 6,8을 모르고 Map을 잘 사용하고 있지 않으셨나요?

Reply:
@ohtaeg 엄청난 길이의!! 피드백에 감사드립니다. StackOverflow에서 피상적으로 느꼈던 감정에 비해서 생각정리에 훨씬 도움이 되었습니다.
특히 ```인터페이스는 계약서, 약속입니다.``` 라는 표현이 인상적이었는데요. 제가 이해하기로는 인터페이스에서 선언했던 상수가 내부 구현체에서 변경이 되는 경우, 사이드 이펙트를 초래할 수 있다는 말씀으로 받아들였습니다. 
상수는 설계와 관련이 되어 있기 때문에 외부로 드러낼 필요도 없다라는 점도 동감이 됩니다. 내부 구현체 쪽으로 몰아주는 것이 메모리 적재 시점에도 논리적으로 맞을 것 같습니다.

이제 저의 궁금증은 다음과 같습니다.
1. 메소드 구현을 강제하는 것처럼 내부 구현체가 특정 상수를 반드시 가지도록 강제할 수 있는 방법이 있겠느냐?
* 내부 구현체가 변경됨에 따라 해당 상수를 사용할 필요가 없게 되는 경우도 있을 것입니다. 하지만 만약 상수의 구현을 강제하고 싶다면(내부 구현체에 따라서 값이 바뀌더라도 괜찮으니) 어떻게 하면 좋을까요. null이나 0으로 초기화하는 것은 답이 아닌 것 같고, 하위 구현체에 대한 강제성도 띄지 않는 것 같더라구요.
2. 공통되는 상수는 별도의 Util 클래스에서 빼서 관리하는 것이 맞을까요? 인터페이스-내부 구현체 간의 관계에서, 별도의 VO에 또 의존해야 하는 것 같아 다소 마음이 불편(??????) 하네요 ㅎㅎ

Reply:
공통되는 상수를 구현체들이 동일하게 강제하는 방법에 대한 의견을 말씀드려볼께요!
- 말씀하신대로 인스턴스화 할 수 없는 Util 클래스에 관리하는 방법
- Enum으로 열거 타입으로 나타내어 여러 타입으로 관리하는 방법
- 추상 클래스를 이용하여 골격 구현 추상 클래스로 관리하는 방법
등이 있을 것 같습니다 :)